### PR TITLE
Change `unnecessary_to_owned` `into_iter` suggestions to `MaybeIncorrect`

### PR DESCRIPTION
**Pull request by smoelius**
I am having a hard time finding a good solution for #8148, so I am wondering if is enough to just change the suggestion's applicability to `MaybeIncorrect`?

I apologize, as I realize this is a bit of a cop out.

changelog: none

### CODE REVIEW

File: clippy_lints/src/methods/unnecessary_iter_cloned.rs [hunk 1]
```diff
@@ -34,6 +34,7 @@ pub fn check_for_loop_iter(
     expr: &'tcx Expr<'tcx>,
     method_name: Symbol,
     receiver: &'tcx Expr<'tcx>,
+    applicability: Applicability,
```
Comment (camsteffen) [inline clippy_lints/src/methods/unnecessary_iter_cloned.rs:37]: Passing applicability to this function is difficult to make sense of. Does this sound right? If it is cloned (or to_owned) *before* `into_iter` is called, that's where we have issues.

```suggestion
    cloned_before_iter: bool,
```

**Comment by camsteffen** [comment]
> I apologize, as I realize this is a bit of a cop out.

No worries! This is one of those very difficult problems where we basically have to reverse-engineer the borrow checker.

File: clippy_lints/src/methods/unnecessary_iter_cloned.rs (hunk 1 above)
Comment (smoelius) [reply clippy_lints/src/methods/unnecessary_iter_cloned.rs:37]: That's right: if there is a call to `clone`/`to_owned` before the call to `into_iter`, the suggestion to remove the `clone`/`to_owned` call could be incorrect, as could follow-on suggestions to remove uses of `.copied()`/`.cloned()` and `&`.

Should the effect of `cloned_before_iter` be the same: to determine the applicability of `check_for_loop_iter`'s suggestions?

Comment (camsteffen) [reply clippy_lints/src/methods/unnecessary_iter_cloned.rs:37]: Yes, purely non-functional change to make the code easier to follow.

Comment (smoelius) [reply clippy_lints/src/methods/unnecessary_iter_cloned.rs]: Following the change, I felt like the comments should be adjusted. I can change them back, if preferred.

**Comment by camsteffen** [comment]
Do you mind adding a "Known Problems" section to the lint docs? Otherwise looks good.

**Comment by smoelius** [comment]
I hope I used the right level of detail.

**Comment by camsteffen** [comment]
Thanks! @bors r±

**Comment by camsteffen** [comment]
That's weird
@bors r+

**Pull request merged**